content/en/opentelemetry/schema_semantics/hostname.md

```diff
@@ -0,0 +1,112 @@
+
+1. Check the general `host` and `datadog.host.name` Datadog-specific conventions.
+1. Cloud provider-specific conventions for AWS, Azure and GCP.
+1. Kubernetes-specific conventions.
+1. Fall back to `host.id` and `host.name` if no specific conventions are found.
+
+The following sections explain each set of specific conventions in detail.
+
+### General hostname semantic conventions
+
+The `host` and `datadog.host.name` conventions are Datadog-specific conventions. They are considered first and can be used to override the hostname detected using the usual OpenTelemetry semantic conventions. Prefer using the `datadog.host.name` convention since it is namespaced and it is less likely to conflict with other vendor-specific behavior.
+
+When using the OpenTelemetry Collector, you can use the `transform` processor to set the `datadog.host.name` convention in your pipelines. For example, to change set the hostname as `my-custom-hostname` in all metrics, traces and logs on a given pipeline, you can use the following configuration:
+
+```yaml
+transform:
+      metric_statements: &statements
+        - context: resource
+          statements:
+            - set(attributes["datadog.host.name"], "my-custom-hostname")
+      trace_statements: *statements # Use the same statements as in metrics
+      log_statements:   *statements # Use the same statements as in metrics
+```
+
+Don't forget to add the `transform` processor to your pipelines.
+
+### Cloud provider-specific conventions
+
+The `cloud.provider` attribute is used to determine the cloud provider. Further attributes are used to determine the hostname for each specific platform. If `cloud.provider` or any of the attributes are missing, the next set of conventions is checked.
+
+#### Amazon Web Services
+
+These conventions are checked if `cloud.provider` has the value `aws`:
+
+1. Check `aws.ecs.launchtype` to determine if the payload comes from an ECS Fargate task. If so, since this is a serverless environment, use `aws.ecs.task.arn` as the identifier with tag name `task_arn`.
+1. Use `host.id` as the hostname otherwise. This matches the EC2 instance id.
+
+#### Google Cloud
+
+These conventions are checked if `cloud.provider` has the value `gcp`:
+
+1. Check that both `host.name` and `cloud.account.id` are available and have the expected format, remove the prefix from `host.name` and merge both into a hostname.
+
+#### Azure
+
+These conventions are checked if `cloud.provider` has the value `azure`:
+
+1. Use `host.id` as the hostname if it is available and has the expected format, fall back to `host.name` otherwise.
+
+### Kubernetes-specific conventions
+
+If `k8s.node.name` and `k8s.cluster.name` are available, the hostname is set to `<node name>-<cluster name>`. If only `k8s.node.name` is available, the hostname is set to the node name.
+
+## Fallback hostname logic
+
+If no valid host names are found, the behavior varies depending on the ingestion path. 
```

Comment:
Before getting here we also check `host.id` and `host.name`, no?

Reply:
Yes, this is mentioned here: https://github.com/DataDog/documentation/blob/290606889a75116feb0a9b34a4cfcd845cf1a324/content/en/opentelemetry/schema_semantics/hostname.md?plain=1#L23 

Should I add a section for this? Maybe right now it's easy to miss

Reply:
Yeah I think adding a section makes sense